src/coreComponents/mesh/generators/VTKUtilities.cpp

```diff
@@ -640,6 +640,115 @@ vtkSmartPointer< vtkDataSet > manageGlobalIds( vtkSmartPointer< vtkDataSet > mes
   return output;
 }
 
+/**
+ * @brief This function tries to make sure that no MPI rank is empty
+ *
+ * @param[in] mesh a vtk grid
+ * @param[in] comm the MPI communicator
+ * @return the vtk grid redistributed
+ */
+vtkSmartPointer< vtkDataSet >
+ensureNoEmptyRank( vtkDataSet & mesh,
+                   MPI_Comm const comm )
+{
+  // step 1: figure out who is a donor and who is a recipient
+  globalIndex const numElems = mesh.GetNumberOfCells();
+  globalIndex const numProcs = MpiWrapper::commSize( comm );
+
+  array1d< globalIndex > elemCounts( numProcs );
+  MpiWrapper::allGather( numElems, elemCounts, comm );
+
+  SortedArray< globalIndex > recipientRanks;
+  array1d< globalIndex > donorRanks;
+  array1d< globalIndex > donorNumElems;
+  recipientRanks.reserve( numProcs );
+  donorRanks.reserve( numProcs );
+  donorNumElems.reserve( numProcs );
+
+  for( globalIndex iRank = 0; iRank < numProcs; ++iRank )
+  {
+    if( elemCounts[iRank] == 0 )
+    {
+      recipientRanks.insert( iRank );
+    }
+    else if( elemCounts[iRank] > 1 ) // need at least two elems to be a donor
+    {
+      donorRanks.emplace_back( iRank );
+      donorNumElems.emplace_back( elemCounts[iRank] );
+    }
+  }
+
+  // step 2: at this point, we need to determine the donors and which cells they donate
+
+  // First we sort the donor in order of the number of elems they contain
+  std::stable_sort( donorRanks.begin(), donorRanks.end(),
+                    [&donorNumElems] ( auto & i1, auto & i2 )
+  { return donorNumElems[i1] < donorNumElems[i2]; } );
```

Comment:
Are you sure this shouldn't just be
```suggestion
  std::stable_sort( donorRanks.begin(), donorRanks.end(),
                    [&elemCounts] ( auto i1, auto i2 )
  { return elemCounts[i1] < elemCounts[i2]; } );
```
since `i1` and `i2` here are both in the range `[0..numProcs)`, not `[0..numDonors)`? If I read this correctly, you won't need `donorNumElems` at all.

Reply:
Thanks for the catch, I fixed it and removed `donorNumElems`